app/controllers/friendships_controller.rb

```diff
@@ -0,0 +1,36 @@
+class FriendshipsController < ApplicationController
+  def index
+    @friends = current_user.all_approved_friends
+  end
+
+  def create
+    @friendship = Friendship.new(friendship_params)
+    if @friendship.save
+      redirect_to :back
+    else
+      flash[:warn] = "Unable to send friend request, please try again"
+      redirect_to :back
+    end
+  end
+
+  def update
+    @friendship = Friendship.find_by(id: params[:id])
+    if @friendship.update_attributes(friendship_params)
+      redirect_to user_path(id: params[:user_id])
+    else
+      flash[:warn] = "There was an error, please try again"
+      redirect_to :back
+    end
+  end
+
+  def destroy
+    @friendship = Friendship.find_by(id: params[:id])
+    @friendship.delete
```

Comment:
Be a little bit careful with delete rather than destroy. If you have any before_destroy etc callbacks defined, calling delete instead of destroy skips them.